android/app/src/main/kotlin/com/jabook/app/jabook/audio/data/repository/ChapterMetadataRepository.kt

```
@@ -68,17 +68,13 @@ public class ChapterMetadataRepository
-
-            @Suppress("UNCHECKED_CAST")
             val chapters =
-                when (result) {
-                    is Result.Success<*> -> (result as Result.Success<List<ChapterMetadataEntity>>).data
-                    is Result.Error -> emptyList()
-                    else -> emptyList()
+                try {
+                    chapterDao
+                        .getChapters(bookId)
+                        .first()
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)